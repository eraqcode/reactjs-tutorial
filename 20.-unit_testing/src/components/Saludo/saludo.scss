.header-saludo {
    margin-top: 40px;
    margin-left: 25px;
    font-size: 1.5rem;
}

.header-saludo h1 {
    margin-left: 12px;
}

.header-saludo h3 {
    margin-top: -30px;
    color: rgb(121, 117, 117);
    font-size: 1.2rem;
}