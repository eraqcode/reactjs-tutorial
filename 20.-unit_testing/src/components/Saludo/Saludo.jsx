import React from 'react'
import './saludo.scss'
import PropTypes from 'prop-types'

export const Saludo = ( { greet, subtitle } ) => {
    return (
        <section>
            <header className="header-saludo">
                <h1> { greet } </h1>
                <h3> { subtitle } </h3>
            </header>
        </section>
    )
}

Saludo.propTypes = {
    greet: PropTypes.string.isRequired,
    subtitle: PropTypes.string.isRequired
}