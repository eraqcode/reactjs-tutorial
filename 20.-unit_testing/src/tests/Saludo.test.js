import { render } from '@testing-library/react'
import React from 'react';
import { Saludo } from '../components/Saludo/Saludo';
import "@testing-library/jest-dom";
import { shallow } from 'enzyme';

describe('Testing in <Saludo /> component', () => {
    
    //test('should show the <Saludo /> component correctly', () => {
    //    
    //    let greet = "Hi, I´m Erik";
    //    let subtitle = "A simple web developer"

    //    const { getByText } = render(
    //        <Saludo 
    //            greet = { greet }
    //            subtitle = { subtitle }
    //        />
    //    )

    //    expect( getByText( greet ) ).toBeInTheDocument();
    //    expect( getByText( subtitle ) ).toBeInTheDocument();   

    //});

    let greet = "Hi, I´m Erik!!!";
        let subtitle = "A simple web developer"

        const wrapper = shallow(
            <Saludo 
                greet = { greet }
                subtitle = { subtitle }
            />
        ) 

    test('show <Saludo /> coponent correctly', () => {
        
        expect( wrapper ).toMatchSnapshot();

    });

    test('show the subtitle', () => {
        
        const h1 = wrapper.find('h1').text().trim();
        const h3 = wrapper.find('h3').text().trim();
        expect(h1).toBe( greet );
        expect(h3).toBe( subtitle );


    });
});