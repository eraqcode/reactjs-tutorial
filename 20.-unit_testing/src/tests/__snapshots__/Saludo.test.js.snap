// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in <Saludo /> component show <Saludo /> coponent correctly 1`] = `
<section>
  <header
    className="header-saludo"
  >
    <h1>
       
      Hi, IÂ´m Erik!!!
       
    </h1>
    <h3>
       
      A simple web developer
       
    </h3>
  </header>
</section>
`;
